### PR TITLE
Bug fixes:

### PR DESCRIPTION
**Pull request by jnpacker**
Jira: https://issues.redhat.com/browse/ACM-1220

1. Include a ManifestWork condition on HypershiftDeployment
2. Fix invalid reason in aws_infra.go
3. For AWS nodePools, include iops: 0
4. Set the Manifest status to `Being configured initially`
5. Fix a problem where targetNamespace did NOT work when NOT using override: MANIFESTWORK
6. Scope the NodePool List query to the targetNamespace (when NOT using override: MANIFESTWORK)
7. Log message on the number of NodePools being processed
8. Change the code flow, we were creating a pull-secret then reading it, and placing it in the ManifestWork, now we get pull-secret from Provider Credential or from the `CLUSTER_NAME`-pull-secret for override: MANIFESTWORK
9. TODO: We should be checking for a ManagedCluster resource when targetManagedCluster is set
10. Add a check for HostedClusterSpec = nil, and report a condition
11. Add a check for spec.credentials.AWS missing when Configure: false (SD hit). Updates a condition.
12. Add ManifestWorkConfigured asExpected condition
13. Bump the Hypershift module version

Signed-off-by: Joshua Packer <jpacker@redhat.com>

### CODE REVIEW

**Comment by jnpacker** [comment]
Jira: https://issues.redhat.com/browse/ACM-1220

File: config/crd/cluster.open-cluster-management.io_hypershiftdeployments.yaml [hunk 1]
```diff
@@ -40,6 +40,10 @@ spec:
       jsonPath: .status.conditions[?(@.type=="PlatformIAMConfigured")].status
       name: Ready
       type: string
+    - description: Reason
+      jsonPath: .status.conditions[?(@.type=="ManifestWorkConfigured")].reason
+      name: MANIFEST
```
Comment (jnpacker) [inline config/crd/cluster.open-cluster-management.io_hypershiftdeployments.yaml]: Maybe I should have called this MANIFESTWORK...

Comment (ianzhang366) [reply config/crd/cluster.open-cluster-management.io_hypershiftdeployments.yaml]: I vote for manifestwork, manifest is a bit too generic and could be understand as the manifest of hypershiftDeployment or it's children resource

Comment (zhujian7) [reply config/crd/cluster.open-cluster-management.io_hypershiftdeployments.yaml]: yes, agree with Ian

File: go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.17
 require (
 	github.com/go-logr/logr v1.2.2
 	github.com/go-logr/zapr v1.2.0
-	github.com/openshift/hypershift v0.0.0-20220308210557-1e81ea19568f
+	github.com/openshift/hypershift v0.0.0-20220322200626-7d5d1ee01dd2
```
Comment (zhujian7) [inline go.mod]: Is this the latest commit for hypershift `release-4.10` branch?

File: api/v1alpha1/hypershiftdeployment_types.go [hunk 3]
```diff
@@ -95,6 +95,7 @@ type HypershiftDeploymentSpec struct {
 	TargetNamespace string `json:"targetNamespace"`
 
 	//TargetManagedCluster specify which managedcluster's namespace the manifestwork would be deployed.
+	//Only applies when Spec.Override: MANIFESTWORK
 	//If not specified, it uses the current namespace.
```
Comment (zhujian7) [inline api/v1alpha1/hypershiftdeployment_types.go]: I think we should delete this line, and note only ManifestWork case needs this field.

Comment (zhujian7) [reply api/v1alpha1/hypershiftdeployment_types.go]: btw, I'm wondering if `targetManagedCluster` is a user-friendly name, I mean we actually use the `management` word in a lot of places, right?

File: pkg/controllers/manifestwork.go [hunk 4]
```diff
@@ -104,19 +106,32 @@ func syncManifestworkStatusToHypershiftDeployment(
...
 
 	m, err := ScaffoldManifestwork(hyd)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
+	// This is a special check to make sure these values are provided as they are Not part of the standard
+	// HostedClusterSpec
+	if hyd.Spec.Credentials == nil ||
+		hyd.Spec.Credentials.AWS == nil {
```
Comment (zhujian7) [inline pkg/controllers/manifestwork.go]: Will this affect the azure case?

File: go.mod (hunk 2 above)
Comment (jnpacker) [reply go.mod]: The two branches are the same, they are turning off FF this week or early next week.

File: api/v1alpha1/hypershiftdeployment_types.go (hunk 3 above)
Comment (jnpacker) [reply api/v1alpha1/hypershiftdeployment_types.go]: We're actually changing ManagementCluster to Hosting Service Cluster from a UX and discussion point of view.

File: go.mod (hunk 2 above)
Comment (jnpacker) [reply go.mod]: But I re-based with the @release-4.10 to be safe.

File: pkg/controllers/manifestwork.go (hunk 4 above)
Comment (jnpacker) [reply pkg/controllers/manifestwork.go]: yes you are correct. Will fixit.

File: api/v1alpha1/hypershiftdeployment_types.go (hunk 3 above)
Comment (jnpacker) [reply api/v1alpha1/hypershiftdeployment_types.go]: Adjusted the txt.

**Comment by zhujian7** [comment]
/lgtm
@jnpacker btw, after this is merged, I think we need to update the CRD in the backplane-operator: https://github.com/stolostron/backplane-operator/blob/main/pkg/templates/crds/hypershift/cluster.open-cluster-management.io_hypershiftdeployments.yaml

**Pull request merged**